ale_linters/sh/shellcheck.vim

```diff
@@ -16,9 +16,21 @@ else
     let s:exclude_option = ''
 endif
 
+function! s:GetDialectArgument()
+    if exists('b:is_bash') && b:is_bash
+        return '-s bash'
+    elseif exists('b:is_sh') && b:is_sh
+        return '-s sh'
+    elseif exists('b:is_kornshell') && b:is_kornshell
+        return '-s ksh'
+    endif
```

Comment:
I've updated this to evaluate multiple times. I haven't added zsh here since it seems like right now this isn't run for zsh files since they have the `zsh` `ft` and not `sh`